lib/makePromise.js

```diff
@@ -895,27 +903,37 @@ define(function() {
 						? process.emit(type, rejection.value, rejection)
 						: process.emit(type, rejection);
 				};
-			} else if(typeof self !== 'undefined' && typeof CustomEvent === 'function') {
-				return (function(noop, self, CustomEvent) {
-					var hasCustomEvent = false;
-					try {
-						var ev = new CustomEvent('unhandledRejection');
-						hasCustomEvent = ev instanceof CustomEvent;
-					} catch (e) {}
-
-					return !hasCustomEvent ? noop : function(type, rejection) {
-						var ev = new CustomEvent(type, {
-							detail: {
-								reason: rejection.value,
-								key: rejection
-							},
-							bubbles: false,
-							cancelable: true
+			}
+			if(typeof self !== 'undefined' && typeof CustomEvent === 'function') {
+				if (hasCustomEvent()) {
```

Comment:
Can this `if(hasCustomEvent())` can be merged into the enclosing `if`, and then the `if` below can become an `else if`?